lib/datasource/docker.js

```diff
@@ -35,66 +49,6 @@ function getRepository(pkgName, registry) {
   return pkgName;
 }
 
-async function getAuthHeaders(registry, repository) {
-  // istanbul ignore if
-  try {
-    const apiCheckUrl = `${registry}/v2/`;
-    const apiCheckResponse = await got(apiCheckUrl, { throwHttpErrors: false });
-    if (apiCheckResponse.headers['www-authenticate'] === undefined) {
-      return {};
-    }
-    const authenticateHeader = new wwwAuthenticate.parsers.WWW_Authenticate(
-      apiCheckResponse.headers['www-authenticate']
-    );
-
-    // prettier-ignore
-    const authUrl = `${authenticateHeader.parms.realm}?service=${authenticateHeader.parms.service}&scope=repository:${repository}:pull`;
-    const { host } = URL.parse(registry);
-    const opts = hostRules.find({ platform: 'docker', host }, { json: true });
-    if (opts.username && opts.password) {
-      const auth = Buffer.from(`${opts.username}:${opts.password}`).toString(
-        'base64'
-      );
-      opts.headers = { Authorization: `Basic ${auth}` };
-    }
-    logger.debug(
-      `Obtaining docker registry token for ${repository} using url ${authUrl}`
-    );
-    const { token } = (await got(authUrl, opts)).body;
-    // istanbul ignore if
-    if (!token) {
-      logger.warn('Failed to obtain docker registry token');
-      return null;
-    }
-    return {
-      Authorization: `Bearer ${token}`,
-      Accept: 'application/vnd.docker.distribution.manifest.v2+json',
-    };
-  } catch (err) /* istanbul ignore next */ {
-    if (err.statusCode === 401) {
-      logger.info(
-        { registry, dockerRepository: repository },
-        'Unauthorized docker lookup'
-      );
-      logger.debug({ err });
-      return null;
-    }
-    if (err.statusCode === 429 && registry.endsWith('docker.io')) {
-      logger.warn({ err }, 'docker registry failure: too many requests');
-      throw new Error('registry-failure');
```

Comment:
@mikebryant code like this is very important for Renovate. We need to make a judgment call on when Docker Hub is failing - which happens often - because if we didn't then we'd end up closing a bunch of PRs only to have to recreate them again the next time Docker Hub is available again.

I think we're missing it in the refactored version?

Reply:
Mhm, fair. I'll see what I can do about pulling this out of the underlying library